### PR TITLE
Network identifier

### PR DESCRIPTION
**Pull request by malroc**
Intended to be a replacement for #174 but is not ready yet. Please consider this PR WIP.

### CODE REVIEW

**Comment by macgregordennis** [comment]
Let's use this approach for #139, which should be ready sometime in the coming week.

File: lib/credo_core_node/application.ex [hunk 1]
```diff
@@ -14,25 +14,8 @@ defmodule CredoCoreNode.Application do
 
     setup_mining_account_storage()
 
-    maybe_load_mining_pk_from_env()
```
Comment (malroc) [inline lib/credo_core_node/application.ex]: Moved to `MineOperator`

File: lib/credo_core_node/application.ex [hunk 2]
```diff
@@ -14,25 +14,8 @@ defmodule CredoCoreNode.Application do
...
-
     maybe_load_genesis_block()
 
-    CredoCoreNode.Mining.calculate_my_miner_status()
```
Comment (malroc) [inline lib/credo_core_node/application.ex]: Moved to `MineOperator`

File: lib/credo_core_node/application.ex [hunk 3]
```diff
@@ -83,17 +71,6 @@ defmodule CredoCoreNode.Application do
     :ets.new(:mining_accounts, [:set, :public, :named_table])
   end
 
-  defp maybe_load_mining_pk_from_env do
```
Comment (malroc) [inline lib/credo_core_node/application.ex]: Moved to `MineOperator`

File: lib/credo_core_node/application.ex [hunk 4]
```diff
@@ -14,25 +14,8 @@ defmodule CredoCoreNode.Application do
 
     setup_mining_account_storage()
 
-    maybe_load_mining_pk_from_env()
-
     maybe_load_genesis_block()
 
-    CredoCoreNode.Mining.calculate_my_miner_status()
-
-    # TODO: migration code, to be removed after moving to testnet B
```
Comment (malroc) [inline lib/credo_core_node/application.ex]: Removed (not needed anymore)

File: lib/credo_core_node/application.ex [hunk 5]
```diff
@@ -57,19 +40,24 @@ defmodule CredoCoreNode.Application do
     :ok
   end
 
-  def root_dir,
-    do: '#{System.user_home()}/credo/#{Application.get_env(:credo_core_node, :environment)}'
+  def root_dir do
```
Comment (malroc) [inline lib/credo_core_node/application.ex]: `root_dir` is now network-dependent, i.e. the same node connected to different networks will use different dirs for storing Mnesia and LevelDB data

File: config/config.exs [hunk 6]
```diff
@@ -15,8 +15,6 @@ config :credo_core_node, CredoCoreNodeWeb.Endpoint,
   render_errors: [view: CredoCoreNodeWeb.ErrorView, accepts: ~w(json)],
   pubsub: [name: CredoCoreNode.PubSub, adapter: Phoenix.PubSub.PG2]
 
-config :credo_core_node, Mnesia, table_suffix: System.get_env("MNESIA_TABLE_SUFFIX") || Mix.env()
```
Comment (malroc) [inline config/config.exs]: We don't need table suffixes anymore as we are no longer using the same Mnesia DB for different environments

**Comment by malroc** [comment]
@macgregordennis should be ready for review/merging now.
The idea is that we don't accept connections from the nodes with different `network_id` (which is passed via `x-ccn-network-id` header in create connection request), i.e. the nodes from different networks don't know each other and don't communicate to each other.
Also, for development/testing simplicity a single node can be configured to connect to different networks via `NETWORK_ID` env; for each network, a separate set of Mnesia and LevelDB databases will be used in this case.
The PR also includes some refactoring/bug fixes.

**Pull request merged**